docs/reference/core/ambassador.md

```diff
@@ -41,6 +41,7 @@ spec:
 | `server_name: envoy` | By default Envoy sets server_name response header to `envoy`. Override it with this variable |  |
 | `service_port: 8080` | If present, service_port will be the port Ambassador listens on for microservice access. If not present, Ambassador will use 8443 if TLS is configured, 8080 otherwise. |  |
 | `statsd` | Configures Ambassador statistics. These values can be set in the Ambassador module or in an environment variable. For more information, see the [Statistics reference](../../statistics#exposing-statistics-via-statsd). |  |
+| `upstream_ambassador_namespace` | Controls whether Ambassador will resolve upstream services assuming they are in the same namespace as the element referring to them, e.g. a Mapping in namespace `foo` will look for its service in namespace `foo`. If `true`, Ambassador will resolve the upstream services assuming they are in the same namespace as Ambassador, unless the service explicitly mentions a different namespace. | `upstream_ambassador_namespace: false` |
```

Comment:
We're changing things such that the upstream service is assumed to be in the same namespace as the element referring to the upstream service so that e.g. a Mapping in namespace foo will look for its service in namespace foo unless the service explicitly mentions a different namespace.

Since this is very definitely a breaking change, we also support a way to turn it off from the ambassador Module: if you set `upstream_ambassador_namespace: true`, then Ambassador will follow the old behavior. The **default will be false**, meaning that Ambassador 1.0 will default to the **new** behavior.

cc. @richarddli , @kelseyevans I think this breaking change is important to highlight in the release blog post.